Firestore/Source/Local/FSTReferenceSet.mm

```diff
@@ -119,15 +124,19 @@ - (FSTDocumentKeySet *)referencedKeysForID:(int)ID {
   return keys;
 }
 
-- (BOOL)containsKey:(FSTDocumentKey *)key {
+- (BOOL)containsKey:(const DocumentKey &)key {
   // Create a reference with a zero ID as the start position to find any document reference with
   // this key.
   FSTDocumentReference *reference = [[FSTDocumentReference alloc] initWithKey:key ID:0];
 
   NSEnumerator<FSTDocumentReference *> *enumerator =
       [self.referencesByKey objectEnumeratorFrom:reference];
-  FSTDocumentKey *_Nullable firstKey = [enumerator nextObject].key;
-  return [firstKey isEqual:reference.key];
+  FSTDocumentReference *_Nullable firstReference = [enumerator nextObject];
+  if (firstReference) {
+    return firstReference.key == reference.key;
+  } else {
+    return false;
+  }
```

Comment:
As above, you can optionally replace lines 135-139 with `return firstReference && firstReference.key == reference.key`